GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -239,46 +255,73 @@ private static void AddTreeNodeToSearchResult(ICollection<TreeNode> ret, TreeNod
 
         private void AddBranches()
         {
-            _branchesTreeRootNode = new TreeNode(Strings.Branches)
+            if (_branchesTree == null)
             {
-                ImageKey = nameof(Images.BranchLocalRoot),
-                SelectedImageKey = nameof(Images.BranchLocalRoot),
-            };
-            _branchesTree = new BranchTree(_branchesTreeRootNode, UICommandsSource, _aheadBehindDataProvider);
+                var rootNode = new TreeNode(Strings.Branches)
+                {
+                    ImageKey = nameof(Images.BranchLocalRoot),
+                    SelectedImageKey = nameof(Images.BranchLocalRoot),
+                };
+                _branchesTree = new BranchTree(rootNode, UICommandsSource, _aheadBehindDataProvider);
+            }
+
             AddTree(_branchesTree, 0);
             _searchResult = null;
         }
 
         private void AddRemotes()
         {
-            _remotesTreeRootNode = new TreeNode(Strings.Remotes)
+            if (_remotesTree == null)
             {
-                ImageKey = nameof(Images.BranchRemoteRoot),
-                SelectedImageKey = nameof(Images.BranchRemoteRoot),
-            };
-            _remotesTree = new RemoteBranchTree(_remotesTreeRootNode, UICommandsSource)
-            {
-                TreeViewNode =
+                var rootNode = new TreeNode(Strings.Remotes)
                 {
-                    ContextMenuStrip = menuRemotes
-                }
-            };
+                    ImageKey = nameof(Images.BranchRemoteRoot),
+                    SelectedImageKey = nameof(Images.BranchRemoteRoot),
+                };
+                _remotesTree = new RemoteBranchTree(rootNode, UICommandsSource)
+                {
+                    TreeViewNode =
+                    {
+                        ContextMenuStrip = menuRemotes
+                    }
+                };
+            }
+
             AddTree(_remotesTree, 1);
             _searchResult = null;
         }
 
         private void AddTags()
         {
-            _tagTreeRootNode = new TreeNode(Strings.Tags)
+            if (_tagTree == null)
```

Comment:
💭  Does it at any point become `null`?
If not, perhaps move the logic to ctor or `OnLoad` ...

Reply:
I cleaned this up. OnUICommandsSourceSet now calls CreateBranches/Remotes/Tags/Submodules, and for show/hide, we use AddTree/RemoveTree.